### ISSUE TITLE
TextField: dismiss keyboard on iOS

### ISSUE THREAD

**Issue by christian-lefty** [flutter/flutter#9188]
For text fields on Android, RETURN dismisses the keyboard and removes focus from the text field.

On iOS there is no such support.

a quick and dirty review of third party iOS apps shows that some apps support RETURN while some don't (Alternatives are: showing a dedicated "exit" button, ensuring some submit button is always above the keyboard, etc...)

For Flutter devs who don't care too much, It would be useful for Flutter to at least offer RETURN as an option (For instance, returnDismissesKeyboard: true, false).

**Comment by eseidelGoogle** [comment]
Posted in channel:

```dart
_focusNode = new FocusNode();
_controller = new TextEditingController();
// on Android the return key closes the keyboard.
// on iOS this is up to the app developper. Since we don't add any "dismiss"
// button or support back from the keyboard etc... we will just implement the same
// behavior as on Android.
//
// This is done by listening to the RETURN key, and losing the focus on the TextField
// when that happens. 
if (defaultTargetPlatform == TargetPlatform.iOS) {
  _controller.addListener(() {
    String currentText = _controller.text;

    if (currentText.isNotEmpty) {
      int code = currentText.codeUnitAt(currentText.length - 1);

      if (code == 10 /* the code for the return key */) {
        // remove the last character, and unfocus.
        _controller.text = currentText.substring(0, currentText.length - 1);
        _focusNode.unfocus();
      }
    }
  });
```

**Comment by eseidelGoogle** [comment]
FYI @cbracken in case your recent changes here changed anything.

**Comment by cbracken** [comment]
Related issue: #9340

**Comment by christian-lefty** [comment]
@cbracken It does sound like #9340 is the same thing yes.

**Comment by cbracken** [comment]
Fixed in flutter/engine#3655 and rolled into the framework in #9861.

Note there's still some followup work on textfields to do -- specifically around enabling configurable return key behaviour on both iOS and Android: #8028 and #9573
